### PR TITLE
Refactor xray quadtree building to use point cloud client.

### PR DESCRIPTION
**Pull request by feuerste**
Previously xrays could only be built from one single octree. Using the point cloud client allows us to specify multiple octree locations and define our own factory.

I tested it by running `build_xray_quadtree` with
- no coloring strategy
- `--coloring_strategy colored_with_intensity`
- `--coloring_strategy colored_with_height_stddev`
- `--coloring_strategy colored`

each on the master branch and this branch.
When running `diff -qr xrays/ xrays_new/` on the corresponding output directories, only `meta.pb` differed (because of the arbitrary ordering of repeated messages), all image files were exactly the same.

### CODE REVIEW

File: point_cloud_client/src/lib.rs [hunk 1]
```diff
@@ -19,12 +21,27 @@ impl PointCloudClient {
                     .map(|octree| *octree)
             })
             .collect();
+        let mut aabb = Aabb3::zero();
+        if let Ok(octrees) = &octrees {
```
Comment (nnmm) [inline point_cloud_client/src/lib.rs]: How about collapsing these ifs with `if let Ok((first, rest)) = octrees().ok().and_then(|octs| oct.split_first())`?

File: point_cloud_client/src/lib.rs [hunk 2]
```diff
@@ -19,12 +21,27 @@ impl PointCloudClient {
...
+            if let Some(first_octree) = octrees.first() {
+                aabb = *first_octree.bounding_box();
+                for octree in octrees.iter().skip(1) {
```
Comment (catevita) [inline point_cloud_client/src/lib.rs]: why does it skip the first element?

File: point_cloud_client/src/lib.rs [hunk 3]
```diff
@@ -19,12 +21,27 @@ impl PointCloudClient {
...
+                    let octree_bbox = octree.bounding_box();
+                    aabb = aabb.grow(octree_bbox.min()).grow(octree_bbox.max());
```
Comment (catevita) [inline point_cloud_client/src/lib.rs]: I also saw recently that there is the method union() to merge two Aabbs

File: point_cloud_client/src/lib.rs (hunk 1 above)
Comment (nnmm) [reply point_cloud_client/src/lib.rs]: Actually, can we use `?` in the expression above and here and bail early?

Comment (feuerste) [reply point_cloud_client/src/lib.rs]: I cleaned up the code, PTAL.

File: point_cloud_client/src/lib.rs (hunk 3 above)
Comment (feuerste) [reply point_cloud_client/src/lib.rs]: Thanks, I modified it.

**Comment by feuerste** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user feuerste. Merge queue now has a length of 1.

**Pull request merged**